File: q0228/q0228.cpp
#include <vector>
#include <string>
using namespace std;

class Solution
{
public:
    vector<string> summaryRanges(vector<int> &nums)
    {
        vector<string> res;
        for (int i = 0; i < nums.size(); i++)
        {
            int start = nums[i];
            while (i + 1 < nums.size() && (nums[i + 1] == nums[i] + 1))
            {
                i++;
            }
            int end = nums[i];

            if (start == end)
                res.push_back(to_string(start));
            else
                res.push_back(to_string(start) + "->" + to_string(end));
        }
        return res;
    }
};

File: q0228/q0228.java
package q0228;

import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<String> summaryRanges(int[] nums) {
        List<String> res = new ArrayList<>();
        for (int i = 0; i < nums.length; i++) {
            int start = nums[i];

            while (i + 1 < nums.length && nums[i + 1] - nums[i] == 1) {
                i++;
            }

            int end = nums[i];
            if (end == start)
                res.add(Integer.toString(start));
            else
                res.add(start + "->" + end);
        }
        return res;
    }
}